src/kirby/components/modal/modal-wrapper/modal-wrapper.component.tns.ts

```diff
@@ -66,4 +88,53 @@ export class ModalWrapperComponent extends ContentView {
     }
     // TODO: by default iOS slides up, however it slides together with the dimmed background, hence needs fixing
   }
+
+  getStatusBarHeight() {
+    var result = 0;
+    if (application.android) {
+      var resourceId = application.android.context
+        .getResources()
+        .getIdentifier('status_bar_height', 'dimen', 'android');
+      if (resourceId) {
+        const screenScale = screen.mainScreen.scale;
+        const pixelValue = application.android.context.getResources().getDimension(resourceId);
+        result = pixelValue / screenScale;
+      }
+    }
+    return result;
+  }
+
+  getNavBarHeight() {
+    if (application.android) {
+      let navBarHeight = 0;
+      let windowManager = application.android.context.getSystemService(
+        android.content.Context.WINDOW_SERVICE
+      );
+      const defaultDisplay = windowManager.getDefaultDisplay();
+
+      const realDisplayMetrics = new android.util.DisplayMetrics();
+      defaultDisplay.getRealMetrics(realDisplayMetrics);
+
+      const realHeight = realDisplayMetrics.heightPixels;
+      const realWidth = realDisplayMetrics.widthPixels;
+
+      const displayMetrics = new android.util.DisplayMetrics();
+      defaultDisplay.getMetrics(displayMetrics);
+
+      const displayHeight = displayMetrics.heightPixels;
+      const displayWidth = displayMetrics.widthPixels;
+
+      if (realHeight - displayHeight > 0) {
+        // Portrait
+        navBarHeight = realHeight - displayHeight;
+      } else if (realWidth - displayWidth > 0) {
+        // Landscape
+        navBarHeight = realWidth - displayWidth;
+      }
+
+      // Convert to device independent pixels and return
+      return navBarHeight / application.android.context.getResources().getDisplayMetrics().density;
```

Comment:
`application.android.context.getResources().getDisplayMetrics().density` can never be 0 ?

Reply:
Not after sdk 17 or something...